lib/utils/config.ts

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2016-2017 Resin.io
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import Promise = require('bluebird');
+import ResinSdk = require('resin-sdk');
+import _ = require('lodash');
+import deviceConfig = require('resin-device-config');
+
+const resin = ResinSdk.fromSharedOptions();
+
+export function generateBaseConfig(application: ResinSdk.Application, options: {}) {
+	options = _.mapValues(options, function(value, key) {
+		if (key === 'appUpdatePollInterval') {
+			return value * 60 * 1000;
+		} else {
+			return value;
+		}
+	});
+
+	return Promise.props({
+		userId: resin.auth.getUserId(),
+		username: resin.auth.whoami(),
+		apiUrl: resin.settings.get('apiUrl'),
+		vpnUrl: resin.settings.get('vpnUrl'),
+		registryUrl: resin.settings.get('registryUrl'),
+		deltaUrl: resin.settings.get('deltaUrl'),
+		pubNubKeys: resin.models.config.getAll().get('pubnub'),
+		mixpanelToken: resin.models.config.getAll().get('mixpanelToken')
```

Comment:
Could we call `getAll()` only once? It seems to send separate requests.

Reply:
True! Good catch, that would've been annoying - now fixed.